modules/wallet/unseeded.go

```diff
@@ -104,36 +158,69 @@ func (w *Wallet) loadSiagKeys(masterKey crypto.TwofishKey, keyfiles []string) er
 	// Drop all unneeded keys.
 	skps = skps[0:skps[0].UnlockConditions.SignaturesRequired]
 
-	// Merge the keys into a single spendableKey.
+	// Merge the keys into a single spendableKey and save it to the wallet.
 	var sk spendableKey
 	sk.UnlockConditions = skps[0].UnlockConditions
 	for _, skp := range skps {
 		sk.SecretKeys = append(sk.SecretKeys, skp.SecretKey)
 	}
-
-	// Create the encrypted spendable key file that gets saved to disk.
-	var skf SpendableKeyFile
-	_, err := rand.Read(skf.UID[:])
-	if err != nil {
-		return err
-	}
-	encryptionKey := uidEncryptionKey(masterKey, skf.UID)
-	plaintextVerification := make([]byte, encryptionVerificationLen)
-	skf.EncryptionVerification, err = encryptionKey.EncryptBytes(plaintextVerification)
+	err := w.loadSpendableKey(masterKey, sk)
 	if err != nil {
 		return err
 	}
-	skf.SpendableKey, err = encryptionKey.EncryptBytes(encoding.Marshal(sk))
+	err = w.saveSettings()
 	if err != nil {
 		return err
 	}
-	w.persist.UnseededKeys = append(w.persist.UnseededKeys, skf)
-	return w.saveSettings()
+	return w.createBackup(filepath.Join(w.persistDir, "Sia Wallet Encrypted Backup - "+persist.RandomSuffix()+settingsFileSuffix))
 }
 
 // LoadSiagKeys loads a set of siag-generated keys into the wallet.
 func (w *Wallet) LoadSiagKeys(masterKey crypto.TwofishKey, keyfiles []string) error {
 	lockID := w.mu.Lock()
 	defer w.mu.Unlock(lockID)
+	err := w.checkMasterKey(masterKey)
+	if err != nil {
+		return err
+	}
 	return w.loadSiagKeys(masterKey, keyfiles)
 }
+
+// Load033xWallet loads a v0.3.3.x wallet as an unseeded key, such that the
+// funds become spendable to the current wallet.
+func (w *Wallet) Load033xWallet(masterKey crypto.TwofishKey, filepath033x string) error {
+	lockID := w.mu.Lock()
+	defer w.mu.Unlock(lockID)
+	err := w.checkMasterKey(masterKey)
+	if err != nil {
+		return err
+	}
+
+	var savedKeys []SavedKey033x
+	err = encoding.ReadFile(filepath033x, &savedKeys)
+	if err != nil {
+		return err
+	}
+	var seedsLoaded int
+	for _, savedKey := range savedKeys {
+		spendKey := spendableKey{
+			UnlockConditions: savedKey.UnlockConditions,
+			SecretKeys:       []crypto.SecretKey{savedKey.SecretKey},
+		}
+		err = w.loadSpendableKey(masterKey, spendKey)
+		if err != nil && err != errDuplicateSpendableKey {
+			return err
+		}
+		if err == nil {
+			seedsLoaded++
+		}
+	}
+	err = w.saveSettings()
+	if err != nil {
+		return err
+	}
+	if seedsLoaded != 0 {
+		return w.createBackup(filepath.Join(w.persistDir, "Sia Wallet Encrypted Backup - "+persist.RandomSuffix()+settingsFileSuffix))
+	}
+	return errAllDuplicates
```

Comment:
You think? I am not certain either way. Seems weird to make a backup if the changes are going to fail in your primary.